streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -799,41 +795,41 @@ public KafkaStreams(final Topology topology,
     public KafkaStreams(final Topology topology,
                         final StreamsConfig config,
                         final Time time) {
-        this(topology.internalTopologyBuilder, config, new DefaultKafkaClientSupplier(), time);
+        this(new TopologyMetadata(topology.internalTopologyBuilder, config), config, new DefaultKafkaClientSupplier(), time);
     }
 
-    private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
-                         final StreamsConfig config,
-                         final KafkaClientSupplier clientSupplier) throws StreamsException {
-        this(internalTopologyBuilder, config, clientSupplier, Time.SYSTEM);
-    }
-
-    private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
+    private KafkaStreams(final Topology topology,
                          final StreamsConfig config,
                          final KafkaClientSupplier clientSupplier,
                          final Time time) throws StreamsException {
+        this(new TopologyMetadata(topology.internalTopologyBuilder, config), config, clientSupplier, time);
+    }
+
+    protected KafkaStreams(final TopologyMetadata topologyMetadata,
```

Comment:
Do these two functions need to be `protected` rather than `private`?

Reply:
Just this one does, as it's called from the child class KafkaStreamsNamedTopologyWrapper. I'll change the one below back to private though